### DISCUSSION TITLE
Bi-Directional Control Using WiCAN?

### DISCUSSION THREAD

**Discussion by Wachobc** [meatpiHQ/wican-fw#141] [General]
Does the WiCAN support bi-directional control of the vehicle under test?

For example, for a diagnostic test requires me to set my fuel rail pressure to maximum.  Will the WiCAN support this functionality?

Thank you,

**Comment by meatpiHQ** [comment]
@Wachobc I'm not sure what you mean by bi-directional control? WiCAN can send an receive frames on the CAN bus.

How do you plan to read the pressure, using PID request or is it broadcasted on the CAN Bus?

**Comment by Wachobc** [comment]
Sorry if I was not clear.  I'm new to this and using the vernacular that the all-in-one scanners use.  

I presume that they are talking about transmitting on the bus to the ECU but I wanted to make sure the WiCAN supported this before I invest in the hardware.  

I believe that your response answered my question.

I believe that the rail pressure is a broadcast message on the bus.

Thank you,
